health-events-analyzer/pkg/reconciler/reconciler.go

```diff
@@ -310,42 +288,49 @@ func (r *Reconciler) validateAllSequenceCriteria(ctx context.Context, rule confi
 	return false, nil
 }
 
-func getFacet(facetName string, timeWindow time.Duration, matchCriteria bson.D) bson.E {
-	return bson.E{
-		Key: facetName,
-		Value: bson.A{
-			bson.D{{Key: "$match", Value: bson.D{
-				{Key: "healthevent.generatedtimestamp.seconds", Value: bson.D{
-					{Key: "$gte", Value: time.Now().UTC().Add(-timeWindow).Unix()},
-				}},
-				{Key: "healthevent.agent", Value: bson.D{{Key: "$ne", Value: "health-events-analyzer"}}},
-			}}},
-			bson.D{{Key: "$match", Value: matchCriteria}},
-			bson.D{{Key: "$count", Value: "count"}},
-		},
+func getPipelineStages(rule config.HealthEventsAnalyzerRule,
+	timeWindow time.Duration,
+	healthEventWithStatus datamodels.HealthEventWithStatus) []map[string]interface{} {
+	seq := rule.Sequence
+
+	pipelineStages := []map[string]interface{}{
+		{"$match": map[string]interface{}{
+			"healthevent.generatedtimestamp.seconds": map[string]interface{}{
+				"$gte": time.Now().UTC().Add(-timeWindow).Unix(),
+			},
```

Comment:
Can the time window match also be done along with the mongodb pipeline in config? I'm thinking all we need to do in code is to add the `healthevent.agent` as a safe guard, everything else can be in config.